connectors/sources/google_cloud_storage.py

```diff
@@ -199,17 +148,114 @@ async def _api_call(
+            "auth_uri": "https://accounts.google.com/o/oauth2/auth",
+            "token_uri": "http://localhost:443/token",
+        }
+        return {
+            "service_account_credentials": {
+                "value": json.dumps(default_credentials),
+                "label": "Google Cloud service account json",
+                "type": "str",
+            },
+            "retry_count": {
+                "value": DEFAULT_RETRY_COUNT,
+                "label": "Maximum retries for failed requests",
+                "type": "int",
+            },
+            "enable_content_extraction": {
+                "value": DEFAULT_CONTENT_EXTRACTION,
+                "label": "Enable content extraction (true/false)",
+                "type": "bool",
+            },
+        }
+
+    async def validate_config(self):
+        """Validates whether user inputs are valid or not for configuration field.
+
+        Raises:
+            Exception: The service account json is empty.
+            Exception: The format of service account json is invalid.
+        """
+        if (
+            self.configuration["service_account_credentials"] == ""
+            or self.configuration["service_account_credentials"] is None
+        ):
+            raise Exception("Google Cloud service account json can't be empty.")
+        try:
+            json.loads(self.configuration["service_account_credentials"])
+        except ValueError:
+            raise Exception("Google Cloud service account is not a valid JSON.")
+
+    def get_storage_client(self):
+        """Initialize and return the GoogleCloudStorageClient
+
+        Returns:
+            GoogleCloudStorageClient: An instance of the GoogleCloudStorageClient.
+        """
+        if self.google_storage_client is not None:
+            return self.google_storage_client
+
+        json_credentials = json.loads(self.configuration["service_account_credentials"])
+
+        if (
+            json_credentials.get("private_key")
+            and "\n" not in json_credentials["private_key"]
+        ):
+            json_credentials["private_key"] = get_pem_format(
+                key=json_credentials["private_key"].strip(),
+                max_split=2,
+            )
+
+        self.google_storage_client = GoogleCloudStorageClient(
+            json_credentials=json_credentials,
+            retry_count=self.configuration["retry_count"],
+        )
+        return self.google_storage_client
+
     async def ping(self):
         """Verify the connection with Google Cloud Storage"""
         if RUNNING_FTEST:
             return
+
         try:
             await anext(
-                self._api_call(
+                self.get_storage_client()._api_call(
                     resource="projects",
                     method="serviceAccount",
                     sub_method="get",
-                    projectId=self.user_project_id,
+                    projectId=self.google_storage_client.user_project_id,
```

Comment:
```suggestion
                    projectId=self.get_storage_client().user_project_id,
```

Let's remove the assumption of the initialization side effect explicitly and always use the getter